### PR TITLE
feat!: soft-delete configuration

### LINKED ISSUE #301: Support for configuring soft-deletes in GCS buckets
### TL;DR

GCP announced a soft-delete feature for GCS here: https://cloud.google.com/resources/storage/soft-delete-announce?hl=en

The announcement claims that support for this feature will be enabled in a promotional period starting today (March 1st).  But there appears to be no support in the terraform module for enabling or disabling it.

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SHA65536**
Closes #301 
I'm new to contributing so I hope I did all that is needed
Tested this on our infra

### CODE REVIEW

File: modules/simple_bucket/variables.tf [hunk 1]
```diff
@@ -150,3 +150,11 @@ variable "public_access_prevention" {
   type        = string
   default     = "inherited"
 }
+
+variable "soft_delete_policy" {
+  description = "Soft delete policies to apply. Format is the same as described in provider documentation https://www.terraform.io/docs/providers/google/r/storage_bucket.html#nested_soft_delete_policy"
+  type = object({
+    retention_duration_seconds = number
+  })
+  default = { retention_duration_seconds = 604800 }
```
Comment (bharathkkb) [inline modules/simple_bucket/variables.tf:159]: So we can default to the provider default

```suggestion
  default = { }
```

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @SHA65536. Could you also update the min provider to 5.22 https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/master/versions.tf#L23

File: modules/simple_bucket/variables.tf (hunk 1 above)
Comment (SHA65536) [reply modules/simple_bucket/variables.tf]: Okay, made it optional for the type to work

**Comment by SHA65536** [comment]
@bharathkkb Let me know if there's something else I need to do? I don't know how to properly update the metadata

**Review by bharathkkb — APPROVED** [review]

**Comment by jdub55** [comment]
What's the reason for min provider 5.22? Can this be backported to a 4.x version of the provider?

**Comment by SHA65536** [comment]
This option does not exist on 4.x and defaults to a retention of 7 days, which will result in additional costs starting august

**Comment by victorcadena** [comment]
@SHA65536 it looks like it does for version 4.85.0, or is this a bug in the docs? https://registry.terraform.io/providers/hashicorp/google/4.85.0/docs/resources/storage_bucket#soft_delete_policy

**Comment by SHA65536** [comment]
@victorcadena Hey, 4.85.0 came out 6 days ago, and they added the soft delete policy to there (2 months after the original pr, and 1 week after my last comment)
That means we no longer need 5.x as minimum version
Feel free to open a PR reducing the minimum version :)

**Comment by victorcadena** [comment]
https://github.com/terraform-google-modules/terraform-google-cloud-storage/pull/321

Let me know if it is that simple or there are more considerations!

**Comment by SHA65536** [comment]
Actually just noticed, the early 5.x versions (until 5.22) don't have soft delete, so I don't know how can we force >4.85 while also having <5.22
Maybe one of the maintainers can decide

**Pull request merged**